Lib/pprint.py

```diff
@@ -525,6 +528,13 @@ def _safe_repr(self, object, context, maxlevels, level):
             return repr(object), True, False
 
         r = getattr(typ, "__repr__", None)
+
+        if issubclass(typ, int) and r is int.__repr__:
+            if self._underscore_numbers:
+                return builtins.format(object, "_d"), True, False
```

Comment:
why not use `f"{object:_d}"` instead of format?  That'll be less bytecode and thus faster.

Reply:
It uses `builtins.format()` according ericvsmith's review (https://github.com/python/cpython/pull/24864#discussion_r593977219): the `builtins.format()` function was considered lightly better than f-string format.

[Format documentation](https://docs.python.org/3/library/string.html#format-examples) shows '%-format' is not the preferred way of formating. `builtins.format()` and f-string format use the same [Format Specification Mini-Language](https://docs.python.org/3/library/string.html#format-specification-mini-language) but there is not an official preference between them (or I did not find it in the documentation). Do you know if there is an official opinion about it? Should I ask on python-dev mailing list?

I'm ready to change to f-string format but I'm afraid another commiter has another opinion about this point and ask to revert it again.

Reply:
f-strings would be best. I was commenting on the built-in `format()` vs `str.format()`.

Reply:
Thank you for the explanation. :)
I committed a change to use f-string format.